### PR TITLE
merge: mobile develop into develop

### PR DESCRIPTION
**Pull request by begonaalvarezd**
> # Description of change
> 
> This PR aims to kickoff mobile support. 
> 
> TODO:
> - [x] Adjust Webpack for mobile, update Capacitor to v3.0 and add support for Capacitor live reload.
> - [x] Kickoff dashboard and child components for mobile, excluding graphs.
> - [x] Kickoff onboarding views and child components for mobile (except migration)
> - [x] Abstract electron and capacitor to platform API
> - [x] Connect iOS bindings
> - [ ] Connect Android bindings
> - [x] Capacitor Pincode Manager
> - [ ] Capacitor Deeplink Manager
> - [ ] Capacitor Notification Manager
> - [ ] ...
> 
> Note: this is a WIP and the development so far is focused on the UI
> 
> ## Links to any relevant issues
> 
> N/A
> 
> ## Type of change
> 
> - New (a change which implements a new feature)
> - Update (a change which updates existing functionality)
> 
> ## How the change has been tested
> 
> In package/mobile run: 
> ```bash
> # in packages/mobile
> yarn
> yarn build
> yarn dev
> ```
> To enable capacitor live reload, keep development server running, then in a new terminal run:
> ```bash
> cp capacitor-sample.config.ts capacitor.config.ts 
> # modify the url field "XXX.XXX.XXX.XXX" with your local IP
> # update project files and open native IDE
> yarn android:update
> yarn ios:update
> ```
> 
> ## Change checklist
> 
> Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.
> 
> - [x] I have followed the contribution guidelines for this project
> - [x] I have performed a self-review of my own code
> - [x] I have commented my code, particularly in hard-to-understand areas
> - [x] I have made corresponding changes to the documentation
> - [ ] I have added tests that prove my fix is effective or that my feature works
> - [ ] I have checked that my latest changes pass CI tests
> - [ ] New and existing unit tests pass locally with my changes
> 

This PR now aims to move mobile develop into the develop branch, so that these two branches don't end up a hard fork with complexities merging.

Tasks are now tracked under the initial https://github.com/iotaledger/firefly/milestone/2

### CODE REVIEW

File: packages/shared/components/Sidebar.svelte [hunk 1]
```diff
@@ -51,24 +62,54 @@
...
-        </button>
-        <span class="flex flex-col items-center">
-            <button class="mb-7 health-status" on:click={() => (showNetwork = true)}>
-                <Icon icon="network" classes="text-{NETWORK_HEALTH_COLORS[healthStatus]}-500" />
-            </button>
-            <button
-                class="w-8 h-8 flex items-center justify-center rounded-full bg-{profileColor}-500 leading-100"
-                on:click={() => (showProfile = true)}>
-                <span class="text-12 text-center text-white uppercase">{profileInitial}</span>
+{#if $mobile}
```
[the remark this answers is not in the document]
Comment (nicole-obrien) [reply packages/shared/components/Sidebar.svelte:65]: To be looked at in the future: https://github.com/iotaledger/firefly/discussions/1601

File: packages/shared/routes/setup/import/views/FileImport.svelte [hunk 2]
```diff
@@ -20,7 +22,7 @@
         dispatch('previous')
     }
 
-    const onDrop = (buffer, name, path) => {
+    const onDrop = (buffer?, name?, path?) => {
```
[the remark this answers is not in the document]
Comment (begonaalvarezd) [reply packages/shared/routes/setup/import/views/FileImport.svelte]: `onFile` is also used on the `on:change` of the mobile input and the `on:click` of the button, if we name it backwards perhaps is also confusing. I'll rename this a bit, let me know if its still confusing

**Comment by nicole-obrien** [comment]
This PR now aims to move mobile develop into the develop branch, so that these two branches don't end up a hard fork with complexities merging.

Tasks are now tracked under the initial https://github.com/iotaledger/firefly/milestone/2

**Review by nicole-obrien — APPROVED** [review]
Let's get this merged, before I have to sort anymore merge conflicts!

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Unable to scroll down on settings page

**Review by nicole-obrien — APPROVED** [review]
Merging before I have to retest this branch. All mobile dev should now be targeting develop.

**Pull request merged**